test/gtest/ucp/test_ucp_ep_reconfig.cc

```diff
@@ -0,0 +1,255 @@
+        sender().connect(&receiver(), get_ep_params());
+        receiver().connect(&sender(), get_ep_params());
+    }
+
+public:
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        add_variant_with_value(variants, UCP_FEATURE_TAG, VARIANT_DEFAULT, "");
+        add_variant_with_value(variants, UCP_FEATURE_TAG,
+                               VARIANT_EXCLUDE_IFACES, "exclude_iface");
+    }
+
+    bool is_exclude_ifaces() const
+    {
+        return get_variant_value() == VARIANT_EXCLUDE_IFACES;
+    }
+
+    void send_recv()
+    {
+        static const unsigned num_iterations = 100;
+        static const size_t msg_size         = 16 * UCS_KBYTE;
+        const ucp_request_param_t param      = {
+            .op_attr_mask = UCP_OP_ATTR_FLAG_NO_IMM_CMPL
+        };
+
+        for (unsigned i = 0; i < num_iterations; ++i) {
+            mem_buffer sbuf(msg_size, UCS_MEMORY_TYPE_HOST, i);
+            mem_buffer rbuf(msg_size, UCS_MEMORY_TYPE_HOST, ucs::rand());
+
+            void *sreq = ucp_tag_send_nbx(sender().ep(), sbuf.ptr(), msg_size,
+                                          0, &param);
+            void *rreq = ucp_tag_recv_nbx(receiver().worker(), rbuf.ptr(),
+                                          msg_size, 0, 0, &param);
+            request_wait(rreq);
+            request_wait(sreq);
+            rbuf.pattern_check(i);
+        }
+    }
+};
+
+void test_ucp_ep_reconfigure::entity::store_config()
+{
+    for (ucp_lane_index_t lane = 0; lane < ucp_ep_num_lanes(ep()); ++lane) {
+        uct_ep_h uct_ep = ucp_ep_get_lane(ep(), lane);
+
+        /* Store UCT endpoints in order to compare with next configuration */
+        if (ucp_wireup_ep_test(uct_ep)) {
+            m_uct_eps.push_back(ucp_wireup_ep(uct_ep)->super.uct_ep);
+        } else {
+            m_uct_eps.push_back(uct_ep);
+        }
+    }
+
+    m_cfg_index = ep()->cfg_index;
+}
+
+ucp_tl_bitmap_t test_ucp_ep_reconfigure::entity::ep_tl_bitmap() const
+{
+    if (ep() == NULL) {
+        return UCS_STATIC_BITMAP_ZERO_INITIALIZER;
+    }
+
+    ucp_tl_bitmap_t tl_bitmap = UCS_STATIC_BITMAP_ZERO_INITIALIZER;
+    for (ucp_lane_index_t lane = 0; lane < ucp_ep_num_lanes(ep()); ++lane) {
+        UCS_STATIC_BITMAP_SET(&tl_bitmap, ucp_ep_get_rsc_index(ep(), lane));
+    }
+
+    return tl_bitmap;
+}
+
+void test_ucp_ep_reconfigure::entity::connect(const ucp_test_base::entity *other,
+                                              const ucp_ep_params_t &ep_params,
+                                              int ep_idx, int do_set_ep)
+{
+    auto r_other     = static_cast<const entity*>(other);
+    auto worker_addr = r_other->get_address(ucp_tl_bitmap_max);
+    ucp_tl_bitmap_t tl_bitmap;
+    unsigned addr_indices[UCP_MAX_LANES];
+    ucp_ep_h ucp_ep;
+
+    tl_bitmap = m_exclude_ifaces ?
+                        UCS_STATIC_BITMAP_NOT(r_other->ep_tl_bitmap()) :
+                        ucp_tl_bitmap_max;

```

Comment:
if the variant is not exclude_ifaces, how is it different than normal connection flow? what is causing a reconfiguration in this case?

Reply:
The idea was to check a flow without reconfig, but I indeed removed it in the next PRs because it is covered by other UCP tests.
I'll backport the removal to this PR.

Reply:
removed variant, but left flag, because will be needed to distinguish from another reconf method in next PR